src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -498,18 +503,14 @@ def dataloader(self) -> Union[TRAIN_DATALOADERS, EVAL_DATALOADERS]:
 
         If the source is a module, the method with the corresponding :attr:`name` gets called.
         """
-        from pytorch_lightning import LightningDataModule, LightningModule  # prevent cyclic import
-
-        if not self.name:
-            return self.instance
```

Comment:
Is this removed on purpose?

Reply:
this is covered by the default return at the end if `self.instance` is neither LM nor LDM.